libsolidity/formal/SMTChecker.cpp

```diff
@@ -853,9 +896,41 @@ void SMTChecker::arithmeticOperation(BinaryOperation const& _op)
 			m_interface->addAssertion(right != 0);
 		}
 
-		checkUnderOverflow(value, intType, _op.location());
+		addOverflowTarget(
+			OverflowTarget::Type::All,
+			_op.annotation().commonType,
+			value,
+			_op.location()
+		);
 
 		defineExpr(_op, value);
+
+		smt::Expression powerOf2 = (0 - minValue(intType)) + maxValue(intType) + 1;
```

Comment:
I would call this ``intValueRange`` or sth like that - it'd work just as well, if it wasn't a power of two. But again: doesn't matter too much.